server/mscalendar/daily_summary.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package mscalendar
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/tz"
+	"github.com/pkg/errors"
+)
+
+const dailySummaryTimeWindow = time.Minute * 2
```

Comment:
If the DailySummaryJob deals with calling the DailySummaryAll every 30 minutes, couldn't we be more generous here? Maybe up to 20 minutes?

Reply:
This is specially important for the (rare) timezones where they have an offset of 15 minutes, instead of a full hour.

Reply:
This time computation algorithm will likely change with the shared timing logic in the cluster job PR. 

Do you think we should enable the users to set their times at 15 minute increments, or keep them at 30 minute increments?

Reply:
If we put them at 15 minutes, we may solve the problem with the timezones, but I have no strong opinion on this.

Reply:
I decided to make the job interval 15 minutes, as well as allow the user to set a time divisible by 15 minutes.

Reply:
(0/5) I still would be a bit worried if the job gets desynchronized and no new summaries being sent (the 2 minute time window seem a bit narrow). Nevertheless, if there is a problem with desynchronization, it does not matter how big is the time window, eventually there will be a problem either way.